app/models/maintenance_tasks/task_data.rb

```diff
@@ -24,48 +24,64 @@ class << self
       #   a Run.
       def find(name)
         task_data = new(name)
-        task_data.last_run || Task.named(name)
+
+        task_data.active_runs.load
+
+        task_data.has_run? || Task.named(name)
+
         task_data
       end
 
       # Returns a list of sorted Task Data objects that represent the
       # available Tasks.
       #
       # Tasks are sorted by category, and within a category, by Task name.
-      # Determining a Task's category require its latest Run record.
-      # To optimize calls to the database, a single query is done to get the
-      # last Run for each Task, and Task Data instances are initialized with
-      # these last_run values.
+      # Determining a Task's category requires their latest Run records.
+      # Two queries are done to get the currently active and completed Run
+      # records, and Task Data instances are initialized with these last_run
+      # values.
       #
       # @return [Array<TaskData>] the list of Task Data.
       def available_tasks
+        tasks = []
+
         task_names = Task.available_tasks.map(&:name)
-        available_task_runs = Run.where(task_name: task_names)
+
+        active_runs = Run.with_attached_csv.active.where(task_name: task_names)
+        active_runs.each do |run|
+          tasks << TaskData.new(run.task_name, run)
+          task_names.delete(run.task_name)
+        end
+
+        completed_runs = Run.completed.where(task_name: task_names)
         last_runs = Run.with_attached_csv.where(
-          id: available_task_runs.select("MAX(id) as id").group(:task_name)
+          id: completed_runs.select("MAX(id) as id").group(:task_name)
         )
-
         task_names.map do |task_name|
           last_run = last_runs.find { |run| run.task_name == task_name }
-          TaskData.new(task_name, last_run)
-        end.sort_by!(&:name)
+          tasks << TaskData.new(task_name, last_run)
+        end
+
+        tasks.sort_by! { |task| [task.name, task.status] }
```

Comment:
Maybe add a comment that this is intended to ensure stable results across database adapters.